adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -164,18 +221,26 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
    *   multiple positions.
    * @see sort
    */
-  def sortLexicographically(): U = {
-    replaceRdd(rdd.sortBy(elem => {
+  def sortLexicographically(
+    partitions: Int = rdd.partitions.length)(implicit tTag: ClassTag[T]): U = {
+
+    val partitionedRDD = rdd.keyBy(elem => {
       val coveredRegions = getReferenceRegions(elem)
 
-      require(coveredRegions.nonEmpty, "Cannot sort RDD containing an unmapped element %s.".format(
-        elem))
+      require(coveredRegions.nonEmpty,
+        "Cannot sort RDD containing an unmapped element %s.".format(elem))
       require(coveredRegions.size == 1,
         "Cannot sort RDD containing a multimapped element. %s covers %s.".format(
           elem, coveredRegions.mkString(",")))
 
       coveredRegions.head
-    }))
+    }).sortBy(elem => elem._1, ascending = true, partitions)
+
+    partitionedRDD.cache()
```

Comment:
I'm conflicted here. On the one hand we use the `optPartitionMap` to determine if a new RDD is sorted, and it's not obvious to me where it should go otherwise (since we have the `replaceRdd()` call in this method). On the other hand, if they have no need for the partition scheme, this is pretty inefficient.